demisto_sdk/commands/doc_reviewer/doc_reviewer.py

```diff
@@ -0,0 +1,372 @@
+import os
+import ssl
+import string
+import sys
+from typing import Dict, Set
+
+import click
+import nltk
+import yaml
+from demisto_sdk.commands.common.constants import FileType
+from demisto_sdk.commands.common.git_util import GitUtil
+from demisto_sdk.commands.common.tools import find_type
+from demisto_sdk.commands.doc_reviewer.known_words import KNOWN_WORDS
+from demisto_sdk.commands.doc_reviewer.rn_checker import ReleaseNotesChecker
+from nltk.corpus import brown, webtext
+from spellchecker import SpellChecker
+
+
+class DocReviewer:
+    """Perform a spell check on the given .yml or .md file.
+    """
+
+    SUPPORTED_FILE_TYPES = [FileType.INTEGRATION, FileType.SCRIPT, FileType.PLAYBOOK, FileType.README,
+                            FileType.DESCRIPTION, FileType.RELEASE_NOTES, FileType.BETA_INTEGRATION,
+                            FileType.TEST_PLAYBOOK, FileType.TEST_SCRIPT]
+
+    def __init__(self, file_path: str, known_words_file_path: str = None, no_camel_case: bool = False,
+                 no_failure: bool = False, expand_dictionary: bool = False, templates: bool = False,
+                 use_git: bool = False, prev_ver: str = None, release_notes_only: bool = False):
+        if templates:
+            ReleaseNotesChecker('', template_examples=True)
+            sys.exit(0)
+
+        # if nothing entered will default to use git
+        elif not file_path and not use_git:
+            use_git = True
+
+        self.file_path = file_path
+        self.git_util = None
+        self.prev_ver = prev_ver if prev_ver else 'demisto/master'
+
+        if use_git:
+            self.git_util = GitUtil()
+
+        if release_notes_only:
+            self.SUPPORTED_FILE_TYPES = [FileType.RELEASE_NOTES]
+
+        self.files = set()  # type:Set
+        self.spellchecker = SpellChecker()
+        self.unknown_words = {}  # type:Dict
+        self.no_camel_case = no_camel_case
+        self.known_words_file_path = known_words_file_path
+        self.found_misspelled = False
+        self.no_failure = no_failure
+        self.expand_dictionary = expand_dictionary
+        self.files_with_misspells = set()  # type:Set
+        self.files_without_misspells = set()  # type:Set
+        self.malformed_rn_files = set()  # type:Set
+
+    @staticmethod
+    def is_camel_case(word):
+        """check if a given word is in camel case"""
+        return word != word.lower() and word != word.upper() and "_" not in word
+
+    @staticmethod
+    def camel_case_split(camel):
```

Comment:
* check out `pascalToSpace` in `CommonServerPython`
* add a unit test

Reply:
done